scripts/GasEstimation.ts

```diff
@@ -0,0 +1,435 @@
+}
+
+async function deployAndSetup() {
+  const owner = Wallet.createRandom().connect(ethers.provider);
+  const penalizer = await deployContract<Penalizer>('Penalizer');
+  const verifier = await deployContract<TestVerifierEverythingAccepted>(
+    'TestVerifierEverythingAccepted'
+  );
+  const token = await deployContract<UtilToken>('UtilToken');
+  const smartWalletTemplate = await deployContract<SmartWallet>('SmartWallet');
+  const relayHub = await deployRelayHub(penalizer.address);
+  await setupRelayHub(relayHub);
+
+  await fundedAccount.sendTransaction({
+    to: owner.address,
+    value: ethers.utils.parseEther('1'),
+  });
+
+  const smartWalletFactory = (await createSmartWalletFactory(
+    smartWalletTemplate,
+    false,
+    owner
+  )) as SmartWalletFactory;
+
+  return {
+    relayHub,
+    owner,
+    smartWalletFactory,
+    token,
+    verifier,
+  };
+}
+
+async function getTransferEstimationWithoutRelay(token: UtilToken) {
+  const [utilAccount] = (await ethers.getSigners()) as [SignerWithAddress];
+
+  await token.mint(TOKEN_AMOUNT_TO_TRANSFER, utilAccount.address);
+
+  const noRelayCall = await token.transfer(
+    utilAccount.address,
+    TOKEN_AMOUNT_TO_TRANSFER
+  );
+
+  return await noRelayCall.wait();
+}
+
+function printRelayGasAnalysis(
+  txReceiptWithRelay: ContractReceipt,
+  txReceiptWithoutRelay: ContractReceipt
+) {
+  const gasOverhead = txReceiptWithRelay.gasUsed.sub(
+    txReceiptWithoutRelay.gasUsed
+  );
+
+  console.log(
+    `\tToken transfer WITHOUT RIFRelay. Gas used: ${txReceiptWithoutRelay.gasUsed.toString()}`
+  );
+  console.log(
+    `\tToken transfer WITH RIFRelay. Gas used:\t ${txReceiptWithRelay.gasUsed.toString()}`
+  );
+  console.log(`\t\t\tGas overhead:\t\t ${gasOverhead.toString()}`);
+}
+
+async function assertRelayedTransaction(
+  forwarderInitialBalance: BigNumber,
+  relayWorkerInitialBalance: BigNumber,
+  balanceToTransfer: BigNumber,
+  token: UtilToken,
+  forwarder: SupportedSmartWallet,
+  fees?: BigNumber
+) {
+  const forwarderFinalBalance = await token.balanceOf(forwarder.address);
+  const relayWorkerFinalBalance = await token.balanceOf(relayWorker.address);
+
+  expect(
+    forwarderInitialBalance.eq(
+      forwarderFinalBalance
+        .add(balanceToTransfer)
+        .add(BigNumber.from(fees ?? 0))
+    )
+  ).to.equal(true, 'SW Payment did not occur');
+  expect(
+    relayWorkerFinalBalance.eq(
+      relayWorkerInitialBalance.add(fees ?? balanceToTransfer)
+    )
+  ).to.equal(true, 'Worker did not receive payment');
+}
+
+function correctEstimatedCallCost(estimation: BigNumber) {
+  const internalCorrection = BigNumber.from(
+    INTERNAL_TRANSACTION_ESTIMATED_CORRECTION
+  );
+  const gasCorrectionFactor = BigNumber.from(ESTIMATED_GAS_CORRECTION_FACTOR);
+
+  return gasCorrectionFactor.mul(
+    estimation.gt(internalCorrection)
+      ? estimation.sub(internalCorrection)
+      : estimation
+  );
+}
+
+async function getCurrentBalances(
```

Comment:
Im not pretty sure what is the benefit of this function.

Reply:
Yes, makes no sense to have this on a function. I just deleted it.